cli/js/web/headers.ts

```diff
@@ -43,15 +46,23 @@ function validateValue(value: string): void {
 class HeadersBase {
   [headerMap]: Map<string, string>;
```

Comment:
This is the wrong structure to store HTTP headers. It should be `Array<[string, string]>`.

Reply:
Using `Array<[string, string]>` would make it browser incompatible. If use call `Headers.prototype.append` with the same key in the browser you expect to override the previous header. But not with `Set-Cookie`. In this case is has to be incompatible with the standardized `Header` implementation. 

IMHO I would prefer HeaderMap to be an `Array` as well to make it work without exceptions. But the user has to be warned, that `append` is not a `Map` anymore.

Reply:
I agree with you, didn't want to change that because of the premise of trying to be fully browser compatible. 

I can do a rework, store headers in `Array<[string, string]>` and change all the methods. But we'll need to define what we'll expect when the user does:

```
header.set('Set-Cooke', 'some-value');
header.set('Set-Cooke', 'other-value');

header.get('Set-Cookie'); // ['some-value', 'other-value']
header.get('Set-Cookie'); // 'some-value, other-value'
```

If we return an array, it's not respecting the specification. Although `Set-Cookie` is not used by users on the client side.

If we use an `Array` I can drop the cookies `Map`, although I'll have to loop every time we try to set a cookie, because we can't store cookies with same name, we should only keep the latest one.

Reply:
And what happens when using an `Array` and `header.set('Content-Type');` twice!

Reply:
> And what happens when using an `Array` and `header.set('Content-Type');` twice!

We can still make it browser compatible by using `Array<[string, string]>` and doing extra validation and looping through the array on every method.  The implementation will be less clean and a bit slower since we'll be iterating constantly through the array.